src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -78,24 +78,23 @@ private bool InitializeBinaryFormatter()
             // If BinaryFormatter support is disabled for the app, the linker will replace this entire
             // method body with "return false;", skipping all reflection code below.
 
-            LazyInitializer.EnsureInitialized(ref s_binaryFormatterType, static () =>
-                Type.GetType("System.Runtime.Serialization.Formatters.Binary.BinaryFormatter, System.Runtime.Serialization.Formatters, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
-                throwOnError: true)!);
+            Interlocked.CompareExchange(
+                ref s_binaryFormatterType,
+                Type.GetType("System.Runtime.Serialization.Formatters.Binary.BinaryFormatter, System.Runtime.Serialization.Formatters, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: true),
```

Comment:
This will run the expensive `Type.GetType` even when `s_binaryFormatterType` is initialized already

Reply:
InitializeBinaryFormatter is only called when _binaryFormatter is null.

Reply:
`_binaryFormatter` is instance field. It needs initialization per each ResourceReader instance. `s_binaryFormatterType` is global state. It needs initializing just once (it is what the original code did at least).

Reply:
It would be nice to structure the initialization as:
```
if (VolatileRead(ref s_binaryFormatterType) is null || Volatile.Read(ref s_deserializeMethod) is null)
{
      // This pattern can be analyzer by IL linker. It will fix the IL linker warnings that the current code hits.
     Type type = GetType(...);
      MethodInfo method = type.GetMethod(...);

      Interlocked.CompareExchange(ref s_binaryFormatterType, ...);
      Interlocked.CompareExchange(ref s_deserializeMethod, ...);
}
```

Reply:
Ugh, you're right, will fix. Thanks.